acceptance: no Docker for remote tests

tbg opened the pull request:
Avoid having to go through `make acceptance` for tests which run against
a remote cluster. Instead, use the usual `make test` invocation (or, if
preferred, `go test` directly.
Triggered by the `-remote` flag. Without that flag, running `./acceptance`
is still a no-op, preserving the desired previous behaviour.

cc @jordanlewis for possible existing test invocations elsewhere that should adapt.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/7560)

<!-- Reviewable:end -->

tbg committed:
acceptance: no Docker for remote tests

Avoid having to go through `make acceptance` for tests which run against
a remote cluster. Instead, use the usual `make test` invocation (or, if
preferred, `go test` directly.
Triggered by the `-remote` flag. Without that flag, running `./acceptance`
is still a no-op, preserving the desired previous behaviour.

tamird commented:
this change doesn't prevent `make acceptance --remote ...` from working, but that command is now for suckers. can you remove this burden of choice?

---

Reviewed 1 of 5 files at r1.
Review status: 1 of 4 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7560#-:-KLXrv8dA5mlnOjRG7oO:-773058677)*
<!-- Sent from Reviewable.io -->

tbg committed:
acceptance: fail on `make acceptance` + remote

Since there is no need for Docker shenanigans, `make test` is the
weapon of choice.

tschottdorf commented:
Done. cc @jordanlewis

---

Review status: 0 of 4 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7560#-:-KLZsW51iFWzSiIABzlv:-843835925)*
<!-- Sent from Reviewable.io -->

cuongdo commented:
Review status: 0 of 4 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[acceptance/allocator_test.go, line 34 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7560#-KLaKalM8OItf4BeeIOB:-KLaKalNpeVzC_bpjYzj:-1440133526) ([raw file](https://github.com/cockroachdb/cockroach/blob/32b32cbcc40febb90c6fe111e21f935bee342dc6/acceptance/allocator_test.go#L34)):*
> ```Go
> //
> // make test \
> //	 TESTTIMEOUT=48h \
> ```

why such a long test timeout for `TestUpreplicate_1To3Small`? the longest observed runtime is 2.5 hours.

---

*[acceptance/allocator_test.go, line 37 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7560#-KLaKw-zKQczVKhMAj8J:-KLaKw-zKQczVKhMAj8K:-1771036783) ([raw file](https://github.com/cockroachdb/cockroach/blob/32b32cbcc40febb90c6fe111e21f935bee342dc6/acceptance/allocator_test.go#L37)):*
> ```Go
> //	 PKG=./acceptance \
> //	 TESTS=Rebalance_3To5Small \
> //	 TESTFLAGS='-v --remote --key-name google_compute_engine --cwd=allocator_terraform --tf.keep-cluster-fail' \
> ```

From looking at:

https://golang.org/pkg/flag/

It looks like a single `-` is the canonical way to specify command-line flags when using the built-in `flag` package.

---

*[acceptance/main_stub_test.go, line 22 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7560#-KLaOKXJQ67MYipZWu8q:-KLaOKXKIHEA5_AmoQKK:-1708444726) ([raw file](https://github.com/cockroachdb/cockroach/blob/32b32cbcc40febb90c6fe111e21f935bee342dc6/acceptance/main_stub_test.go#L22)):*
> ```Go
> // tag to separate invocations of `go test` which are intended to run the
> // acceptance tests from those which are not. The corollary file to this one
> // is main_test.go
> ```

append a comment here regarding the use of this file for remote tests?

---

*[acceptance/main_test.go, line 34 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7560#-KLaMULlYvAI9MrX8M5A:-KLaMULlYvAI9MrX8M5B:-1456465896) ([raw file](https://github.com/cockroachdb/cockroach/blob/32b32cbcc40febb90c6fe111e21f935bee342dc6/acceptance/main_test.go#L34)):*
> ```Go
> func TestMain(m *testing.M) {
> 	if *flagRemote {
> 		fmt.Println("use `make test [...]` instead of `make acceptance [...]` when running remote cluster")
> ```

errors should probably go to `os.Stderr`?

---

*[acceptance/util_test.go, line 83 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7560#-KLaM6FXOKRN6-DhWjQR:-KLaM6FXOKRN6-DhWjQS:-1590103679) ([raw file](https://github.com/cockroachdb/cockroach/blob/32b32cbcc40febb90c6fe111e21f935bee342dc6/acceptance/util_test.go#L83)):*
> ```Go
> 	randutil.SeedForTests()
> 	go func() {
> 		sig := make(chan os.Signal, 1)
> ```

I know that this change is merely code movement, but could you add a comment regarding the intent of this goroutine? It's not obvious from a quick glance.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7560)*
<!-- Sent from Reviewable.io -->

cuongdo commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> with some comments

---

Review status: 0 of 4 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7560#-:-KLaOTTPGTWsxLP0zUs3:-1969120395)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Review status: 0 of 4 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[acceptance/allocator_test.go, line 34 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7560#-KLaKalM8OItf4BeeIOB:-KLurKWJ2L11aU9CrIWd:395317979) ([raw file](https://github.com/cockroachdb/cockroach/blob/32b32cbcc40febb90c6fe111e21f935bee342dc6/acceptance/allocator_test.go#L34)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary>
> why such a long test timeout for `TestUpreplicate_1To3Small`? the longest observed runtime is 2.5 hours.
</details>
We effectively want to disable timeouts when we run these tests, no? Would be a shame to put 3 hours here and then someone uses this for one of the bigger tests (which might as well be a day).

---

*[acceptance/allocator_test.go, line 37 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7560#-KLaKw-zKQczVKhMAj8J:-KLus79EiFXQ4ZFeoQPV:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/32b32cbcc40febb90c6fe111e21f935bee342dc6/acceptance/allocator_test.go#L37)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary>
> From looking at:
> 
> https://golang.org/pkg/flag/
> 
> It looks like a single `-` is the canonical way to specify command-line flags when using the built-in `flag` package.
</details>
Done.

---

*[acceptance/main_stub_test.go, line 22 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7560#-KLaOKXJQ67MYipZWu8q:-KLusUTwCk0_EiHVxyHD:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/32b32cbcc40febb90c6fe111e21f935bee342dc6/acceptance/main_stub_test.go#L22)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary>
> append a comment here regarding the use of this file for remote tests?
</details>
Done.

---

*[acceptance/main_test.go, line 34 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7560#-KLaMULlYvAI9MrX8M5A:-KLutDysgB9xnu1huaeC:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/32b32cbcc40febb90c6fe111e21f935bee342dc6/acceptance/main_test.go#L34)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary>
> errors should probably go to `os.Stderr`?
</details>
Done.

---

*[acceptance/util_test.go, line 83 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7560#-KLaM6FXOKRN6-DhWjQR:-KLuu4FfIuzhxn9c-wlg:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/32b32cbcc40febb90c6fe111e21f935bee342dc6/acceptance/util_test.go#L83)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary>
> I know that this change is merely code movement, but could you add a comment regarding the intent of this goroutine? It's not obvious from a quick glance.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7560)*
<!-- Sent from Reviewable.io -->
